codegen/smithy-aws-typescript-codegen/src/main/java/software/amazon/smithy/aws/typescript/codegen/AddAwsAuthPlugin.java

```diff
@@ -206,19 +206,19 @@ public void writeAdditionalFiles(
         String noTouchNoticePrefix = "// Please do not touch this file. It's generated from template in:\n"
                 + "// https://github.com/aws/aws-sdk-js-v3/blob/main/codegen/smithy-aws-typescript-codegen/"
                 + "src/main/resources/software/amazon/smithy/aws/typescript/codegen/";
-        writerFactory.accept("defaultRoleAssumers.ts", writer -> {
+        writerFactory.accept("src/defaultRoleAssumers.ts", writer -> {
```

Comment:
Why not use CodegenUtils.SOURCE_FOLDER here?

Reply:
Same as other places in this file. 

Also should we use path.join() equivalent instead of concatenate strings with "/"?

Reply:
> Why not use CodegenUtils.SOURCE_FOLDER here?

It was a miss. Added in [`7b69b9a` (#2844)](https://github.com/aws/aws-sdk-js-v3/pull/2844/commits/7b69b9aac097872f151fb6287683d421b7c0ef52)

> Also should we use path.join() equivalent instead of concatenate strings with "/"?

Agree. Created a utility to discuss utility to be used for joining paths at https://github.com/awslabs/smithy-typescript/issues/436